02-sql-aggregation.md

```diff
@@ -114,32 +114,121 @@ think of views is as a table, that can read, aggregate, and filter information
 from several places before showing it to you.
 
 Creating a view from a query requires to add `CREATE VIEW viewname AS`
-before the query itself. For example, if we want to save the query giving
-the number of individuals in a view, we can write
+before the query itself. For example, imagine that my project only covers
+the data gathered during the summer (May - September) of 2000.  That 
+query would look like: 
 
-    CREATE VIEW species_count AS
-    SELECT species_id, COUNT(*)
+    SELECT *
     FROM surveys
-    GROUP BY species_id;
+    WHERE year == 2000 AND (month > 4 AND month < 10)
 
-Now, we will be able to access these results with a much shorter notation:
+But we don't want to have to type that every time we want to ask a 
+question about that particular subset of data!  Let's create a view: 
 
+    CREATE VIEW summer_2000 AS
     SELECT *
-    FROM species_count;
-
-Assuming we do not need this view anymore, we can remove it from the database
-almost as we would a table:
-
-    DROP VIEW species_count;
+    FROM surveys
+    WHERE year == 2000 AND (month > 4 AND month < 10)
 
 You can also add a view using *Create View* in the *View* menu and see the
 results in the *Views* tab just like a table
 
+Now, we will be able to access these results with a much shorter notation:
+
+    SELECT *
+    FROM summer_2000;
+
 > ### Challenge
 >
 > Write a query that returns the number of each species
 caught in each year sorted from most often caught species to the least
 occurring ones within each year starting from the most recent records. Save
 this query as a `VIEW`.
 
+## Null values
+
+Using the view we created in the previous section (`summer_2000`), let's 
+talk about null values.  Missing values in SQL are identified with the 
+special NULL value.  Scroll through our `summer_2000` view.  It should be 
+easy to find several records with missing values.  How do you think we 
+could filter to find these rows?  
+
+You might want to try a query like: 
+
+    SELECT *
+    FROM summer_2000
+    WHERE species_id == NULL
```

Comment:
I don't think it is wise to start with something that is not working. Cut this example and keep the next one.